tools/topology/sof/pipe-eq-fir-volume-playback.m4

```diff
@@ -52,7 +52,7 @@ C_CONTROLBYTES(EQFIR, PIPELINE_ID,
 W_PCM_PLAYBACK(PCM_ID, Passthrough Playback, 2, 0, 2)
 
 # "Volume" has 2 source and 2 sink periods
-W_PGA(0, PIPELINE_FORMAT, 2, 2, 2, LIST(`		', "Master Playback Volume PIPELINE_ID"))
+W_PGA(0, PIPELINE_FORMAT, 2, 2, 2, LIST(`		', "PIPELINE_ID Master Playback Volume"))
```

Comment:
@lgirdwood let me test that and get back

Reply:
@zhuyingjiang @lgirdwood I have a suggestion to fix this problem and improve the naming so alsamixer picks up the direction properly. 

Today, we're using source as Master for all pipelines. Instead I think we should fix the source to something more meaningful like Speaker or Headphone or HDMI etc. Then we we can have names like "Speaker Playback Volume". This way we can avoid the pipeline number in the kcontrol name after the function and confuse alsamixer and would read a lot better as a name too. For our reference alsamixer also shows the widget name to which the kcontrol is attached like PGA3.0 which has the pipeline information already. 

What do you think?

Reply:
@ranj063 Could you share in "PGA1.0 1 Master Playback Volume" what PGA1.0 means? I think currently we can add e.g "Speaker" "Headphone" "HDMI" before Master. so "Speaker Master Playback Volume" works well. If using Speaker replace Master, we need change alsamixer? @bardliao Could you share your point?

Reply:
this fix [https://github.com/thesofproject/linux/issues/43](url)

Reply:
@zhuyingjiang I made the suggestion based on 
https://people.freedesktop.org/~narmstrong/meson_drm_doc/sound/designs/control-names.html

Does replacing "Master" by "Speaker" give errors?

Reply:
@zhuyingjiang PGA1.0 means...the kcontrol is attached to the PGA widget with ID 0 in pipeline 1.

Reply:
@bardliao totally agree. I was sighting examples based on what the pcm names were defined as in sof-glk-da7219.m4. But we should assign the same name for the source as the PCM name in the corresponding pipeline.

Reply:
> Just to confirm, did you try SOURCE : DIRECTION: FUNCTION ? As it reads better in English ?
@lgirdwood Yes, we are <DETAIL> SOURCE : DIRECTION: FUNCTION, and <DETAIL> SOURCE became the name of the control display in alsamixer.

Reply:
@zhuyingjiang "Speaker Playback Volume" should work fine.

Reply:
@ranj063 @bardliao Yes, it works, I mean if "Speaker Master Playback Volume" better than "Speaker  Playback Volume"

Reply:
@zhuyingjiang No need to add "Master" to source. Could you please modify the PR to add the PCM name as the source name for all the kcontrols?

Reply:
@ranj063 OK, will do that.

Reply:
@zhuyingjiang once the new updates are ready, can you change commit messages and PR message to reflect the new format. Thanks !

Reply:
@lgirdwood OK, will do that, thanks for inform.

Reply:
> @zhuyingjiang PGA1.0 means...the kcontrol is attached to the PGA widget with ID 0 in pipeline 1.
@ranj063 if PGA1.0 already include the pipeline ID, is it can be omit when we have name like "PGA6.0 6 Master"? so we can define:
W_PGA(0, PIPELINE_FORMAT, 2, 2, 2, LIST(`              ', "Master Playback Volume")
instead of 
W_PGA(0, PIPELINE_FORMAT, 2, 2, 2, LIST(`              ', "PIPELINE_ID Master Playback Volume")

Reply:
@zhuyingjiang I dont understand your question. Are you asking what to use for the kcontrol name in the W_PGA macro?
Then, it should match the name that you have in the mixer control macro (ie the PCM name followed by Playback and Volume). The PCM name should come from the top-level m4 file for ex: sof-glk-da7219.m4. Look for the PCM name in the PCM_PLAYBAK_ADD/PCM_CAPTURE_ADD/PCM_DUPLEX_ADD macros.

Reply:
@ranj063 Let's give a example: the name before is like "PGA1.0 Master Playback Volume 1", the "1" is pipeline ID, and the "0" is the index. The alsamixer groups the name with the last two words, so this is why this bug occurs. If we move the last "1" before Master, it become "PGA1.0 1 Master Playback Volume", alsamixer works properly(can separate play/capture by F3 F4), the display name is "PGA1.0 1 Master" and when we remove the last "1", is also works properly, and display the name of "PGA1.0 Master".
     By your suggestion, you mean replace the word "Master" with "MIC" "Speaker" or sth else, so the name should be like "PGA1.0 Speaker Playback Volume". Am I understand right?

Reply:
@zhuyingjiang I think @ranj063 's suggestion is replace "PGA1.0 Master" with "Speaker" where "Speaker" is the name of PCM. So, the kcontrol name will be "Speaker Playback Volume". @ranj063 Please correct me if I was wrong.

Reply:
@zhuyingjiang @bardliao yes thats correct.

Reply:
ok, any update to be merged ?

Reply:
@lgirdwood Will update soon, I am now working on WHL power management

Reply:
@ranj063 @lgirdwood 
Just found that the PGAX.Y can't be removed, the firmware need the name to add the widget, and the name must be distinct. or else will have the error :
 [ 6.556897] sof-audio sof-audio: tplg: load kcontrol index 35 chans 2
[ 6.556900] sof-audio sof-audio: tplg: ready widget id 35 pipe 12 type 6 name : PGA stream none
[ 6.556903] sof-audio sof-audio: error: invalid kcontrol count 8 for volume
[ 6.556964] sof-audio sof-audio: error: DSP failed to add widget id 0 type 6 name : PGA stream none reply 0
[ 6.556968] sof-audio sof-audio: ASoC: failed to load widget PGA
[ 6.556971] sof-audio sof-audio: error: tplg component load failed -22
[ 6.556992] sof-audio sof-audio: error: failed to load DSP topology -22
[ 6.557166] sof-audio sof-audio: ASoC: failed to probe component -22
[ 6.557188] sof-nocodec sof-nocodec: ASoC: failed to instantiate card -22
[ 6.557319] sof-nocodec: probe of sof-nocodec failed with error -22
[ 6.557328] sof-audio sof-audio: created machine sof-nocodec
@ranj063 @bardliao Any idea?

Reply:
@zhuyingjiang if you are changing widget names in one place you need to make sure the name is changed in the places that also reference the widget.

Reply:
@lgirdwood Yes, definitely. But the widget name should be distinct, that means if we have two widget have the same name, then the reference can't find the widget, and also the mixer.
We should note when we add a top m4, then the name given to the widget must be distinct.